utilities/pipelines/staticValidation/compliance/module.tests.ps1

```diff
@@ -271,6 +271,39 @@ Describe 'File/folder tests' -Tag 'Modules' {
                 $pathExisting | Should -Be $true -Because "path [$filePath] is expected to exist."
             }
         }
+
+        It '[<moduleFolderName>] Resource Modules must not skip the "defaults" or "waf-aligned" tests with a [` .e2eignore `] file.' -TestCases ($topLevelModuleTestCases | Where-Object { $_.moduleType -eq 'res' }) {
+
+            param(
+                [string] $moduleFolderName,
+                [string] $moduleFolderPath
+            )
+
+            $e2eTestFolderPathList = Get-ChildItem -Directory (Join-Path -Path $moduleFolderPath 'tests' 'e2e') | Where-Object { $_.Name -in 'defaults', 'waf-aligned' }
+            foreach ($e2eTestFolderPath in $e2eTestFolderPathList) {
+                $filePath = Join-Path -Path $e2eTestFolderPath '.e2eignore'
+                $pathExisting = Test-Path $filePath
+                $pathExisting | Should -Be $false -Because 'skipping this test is not allowed.'
+            }
+        }
+
+        It '[<moduleFolderName>] a [` .e2eignore `] file must contain text, which states the exception reason for not executing the deployment test.' -TestCases $topLevelModuleTestCases {
+
+            param(
+                [string] $moduleFolderName,
+                [string] $moduleFolderPath
+            )
+
+            $e2eTestFolderPathList = Get-ChildItem -Directory (Join-Path -Path $moduleFolderPath 'tests' 'e2e')
+            foreach ($e2eTestFolderPath in $e2eTestFolderPathList) {
```

Comment:
Please note that this test will fail for the very first wrong case and not display the others. As discussed in a different location (most recently with the telemetry tests), please collect the wrong cases in a list and then test that the list is empty.